Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs

```diff
@@ -460,6 +491,44 @@ public ActionResult Details(int? id)
             return this.View(problem);
         }
 
+        /// <summary>
+        /// Creates a zip file with all additional files in the problem
+        /// </summary>
+        /// <param name="id">Problem id</param>
+        /// <returns>Zip file containing all additional files</returns>
+        public ActionResult DownloadAdditionalFiles(int id)
+        {
+            var problem = this.Data.Problems.All().FirstOrDefault(x => x.Id == id);
+
+            if (problem == null)
+            {
+                this.TempData.AddDangerMessage(GlobalResource.Problem_not_found);
+                return this.RedirectToAction(GlobalConstants.Index);
+            }
+
+            if (!this.CheckIfUserHasProblemPermissions(id))
+            {
+                this.TempData[GlobalConstants.DangerMessage] = "Нямате привилегиите за това действие";
```

Comment:
Please be consistent in your code. Use only one of the followings:

- `this.TempData.AddDangerMessage("message");`
- `this.TempData[GlobalConstants.DangerMessage] = "message";`